### PR TITLE
Bug #6822. Provided maxSize property for the sidebar and added a check in resizer.js to convert maxSize into pixels if provided in percentage.

### PR DESCRIPTION
**Pull request by prateekagr98**
Provide maxSize to be 45% of window width if element is side bar and maxSize is not specified

This is for #6822.

### CODE REVIEW

**Comment by ficristo** [comment]
@redmunds do you know if this is still relevant?

**Comment by redmunds** [comment]
@ficristo Issue #6822 is still open, so I think this is still relevant. The contributor seems to have addressed all comments (although didn't add a "ready for another review" comment), but I have not verified the fix.

One change I would like to see is to have the max be larger (something like 80%). It might not seem useful to only show 20% of window for editor, but you might want to temporarily expand to find a particular file.

File: src/htmlContent/main-view.html [hunk 1]
```diff
@@ -40,7 +40,7 @@
 
     <!-- Main UI: horizontal set of sidebar, main content vertical stack, and vertical toolbar -->
     <div class="main-view">
-        <div id="sidebar" class="sidebar panel quiet-scrollbars horz-resizable right-resizer collapsible" data-minsize="0" data-forceleft=".content">
+        <div id="sidebar" class="sidebar panel quiet-scrollbars horz-resizable right-resizer collapsible" data-minsize="0" data-maxsize="45%" data-forceleft=".content">
```
Comment (redmunds) [inline src/htmlContent/main-view.html]: I think max should be large. Maybe 80% ?

**Comment by ficristo** [comment]
Thank you @redmunds
@prateekagr98 are you still interested? Could you update with the suggestion of @redmunds?

**Comment by ficristo** [comment]
Thank you for contributing to Brackets.
We committed a different fix in #13026

**Pull request closed without merging**